crates/matrix-sdk-ui/src/timeline/inner/mod.rs

```diff
@@ -117,12 +122,84 @@ impl Default for TimelineInnerSettings {
     fn default() -> Self {
         Self {
             track_read_receipts: false,
-            event_filter: Arc::new(|_| true),
+            event_filter: Arc::new(default_event_filter),
             add_failed_to_parse: true,
         }
     }
 }
 
+/// The default event filter for [`TimelineInnerSettings::event_filter`].
+///
+/// It filters out events that are never visible by default.
+fn default_event_filter(event: &AnySyncTimelineEvent) -> bool {
```

Comment:
Wow, this filter is much more complicated than I expected. Can we make it public somewhere so that users can use the default filter and exclude additional items easily? Otherwise this would be a real pain for Fractal, I think (cc @zecakeh).

Reply:
Yeah, basically we have to match what `TimelineEventHandler::handle_event(TimelineEventKind::from_event(event))` does, hence the complexity; and it makes it harder to test too.

Reply:
Why not apply an inner filter on the value at the output of the user's filter instead? Make sure that  events that are never added will always return `false` and events that are always added will always return `true`, and the other events return the value from the user's filter.

That way it's sure to always work correctly.

Reply:
Alright, added better documentation and made the event filter public now :+1: